praw/models/util.py

```diff
@@ -13,19 +14,23 @@ class BoundedSet:
     def __init__(self, max_items: int):
```

Comment:
Should this still be named `BoundedSet` since it doesn't utilize a `set` object anymore?

Reply:
I mean the dictionary operates similar to a set, and since only the key portion is utilized, it means that the class is functionally still a set.

Reply:
I understand that, however, before it actually utilized methods that are only present in the `set` object as seen here: https://github.com/praw-dev/praw/blob/742e61001e5eafbdd72eca5d8ace0f925c954533/praw/models/util.py#L23-L28

With this change, it no longer utilizes `set` methods and instead uses subscription which isn't possible with the base `set` object.

Reply:
Changing the name would be a breaking change, so any name changes would have to occur in PRAW 8. Furthermore, it's a container, so `BoundedContainer` would make the most sense for a new name.

Reply:
I'd like to call it a set because it implements a subset of the set interface (modulo the size bound, of course). It doesn't use a `set` anymore, but that's an implementation detail.

Reply:
I don't feel a rename is a breaking change as it is intended to be used by the `stream_generator` method. If it was intended to be widely used then there needs to be examples documented.

Reply:
BoundedSet does have documentation, found here: https://praw.readthedocs.io/en/latest/code_overview/other/util.html?highlight=BoundedSet#praw.models.util.BoundedSet

Reply:
I mean code examples.

Reply:
> I'd like to call it a set because it implements a subset of the set interface (modulo the size bound, of course). It doesn't use a `set` anymore, but that's an implementation detail.

Fair enough.

Reply:
> I mean code examples.

If it exists in the documentation then there's a greater-than-zero chance someone was using it, and since it's not marked as a private class for internal use only (by being openly documented and searchable), it is more likely than not to break something.